frontend_app_file/src/container/PopupCreateFile.jsx

```diff
@@ -72,74 +64,60 @@ class PopupCreateFile extends React.Component {
     }
   })
 
-  handleUploadEnd = () => {
+  handleAllFileUploadEnd = (uploadedFileList) => {
     const { state, props } = this
 
-    const uploadedFileFailedList = state.uploadedFileList.filter(f => f.jsonResult.code !== 200)
+    const uploadedFileFailedList = uploadedFileList.filter(f => f.jsonResult.code !== 200)
 
     GLOBAL_dispatchEvent({ type: CUSTOM_EVENT.REFRESH_CONTENT_LIST, data: {} })
 
     if (uploadedFileFailedList.length === 0) {
-      if (state.uploadFileList.length === 1) {
+      if (state.fileToUploadList.length === 1) {
         GLOBAL_dispatchEvent({
           type: CUSTOM_EVENT.OPEN_CONTENT_URL,
           data: {
-            workspaceId: state.uploadedFileList[0].jsonResult.workspace_id,
+            workspaceId: uploadedFileList[0].jsonResult.workspace_id,
             contentType: state.appName,
-            contentId: state.uploadedFileList[0].jsonResult.content_id
+            contentId: uploadedFileList[0].jsonResult.content_id
           }
         })
       }
-      this.handleClose()
-    } else {
-      this.sendGlobalFlashMessage(props.t("Some file(s) couldn't be uploaded"))
-      this.setState({
-        uploadFileList: uploadedFileFailedList,
-        uploadedFileList: [],
-        uploadStarted: false
-      })
+      this.handleClose(false)
+      return
     }
+
+    this.sendGlobalFlashMessage(props.t("Some file(s) couldn't be uploaded"))
+    this.setState({
+      fileToUploadList: uploadedFileFailedList,
+      uploadStarted: false
+    })
   }
 
   handleChangeFile = newFileList => {
     const { props, state } = this
 
     if (!newFileList || newFileList.length === 0) return
 
-    if (newFileList.length === 1 && state.uploadFileList.length === 0) {
+    if (newFileList.length === 1 && state.fileToUploadList.length === 0) {
       this.loadUploadFilePreview(newFileList[0])
-    } else if (state.uploadFilePreview) this.setState({ uploadFilePreview: false })
-
-    const alreadyUploadedList = newFileList.filter(newFile => state.uploadFileList.some(stateFile => stateFile.name === newFile.name))
-    if (alreadyUploadedList.length) {
-      GLOBAL_dispatchEvent({
-        type: CUSTOM_EVENT.ADD_FLASH_MSG,
-        data: {
-          msg: <div>{props.t('Files already uploaded:')}<br /><ul>{alreadyUploadedList.map(file => <li>{file.name}</li>)}</ul></div>,
-          type: 'warning',
-          delay: undefined
-        }
-      })
-      return
+    } else if (state.uploadFilePreview) this.setState({ uploadFilePreview: PREVIEW_POSSIBLE_CASE.NO_PREVIEW })
+
+    const alreadyUploadedList = newFileList.filter((newFile) => this.filesAlreadyInListFilter(newFile, state.fileToUploadList))
+    if (alreadyUploadedList.length > 0) {
+      this.sendGlobalFlashMessage(<div>{props.t('Files already uploaded:')}<br /><ul>{alreadyUploadedList.map(file => <li>{file.name}</li>)}</ul></div>)
+      newFileList = newFileList.filter((newFile) => !this.filesAlreadyInListFilter(newFile, state.fileToUploadList))
```

Comment:
never reassign a parameter.
Create a new variable with a name that reflects what the filter is doing